server/middleware/validationCheck.js

```diff
@@ -0,0 +1,9 @@
+const asyncHandler = require("express-async-handler");
+const { validationResult } = require("express-validator");
+
+exports.checkForValidationErrors = asyncHandler(async (req, res, next) => {
+  const errors = validationResult(req);
+  if (!errors.isEmpty()) return next(errors);
+
+  return next();
+});
```

Comment:
I see what you were getting at with using custom middleware to DRY up the error checking in the controllers!